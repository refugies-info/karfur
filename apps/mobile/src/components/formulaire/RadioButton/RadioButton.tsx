import React from "react";
import { Icon } from "react-native-eva-icons";
import { useTheme } from "styled-components/native";

export interface RadioButtonProps {
  isSelected: boolean;
}

export const RadioButton = (props: RadioButtonProps) => {
  const theme = useTheme();
  return (
    <Icon
      name={props.isSelected ? "radio-button-on" : "radio-button-off"}
      width={24}
      height={24}
      fill={
        props.isSelected ? theme.colors.dsfr_action : theme.colors.dsfr_dark
      }
    />
  );
};

export default RadioButton;
