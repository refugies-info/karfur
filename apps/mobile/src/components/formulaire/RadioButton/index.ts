export { default, RadioButtonProps } from "./RadioButton";
