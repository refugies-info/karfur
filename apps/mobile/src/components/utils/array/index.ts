export { default as isLastChild } from "./isLastChild";
