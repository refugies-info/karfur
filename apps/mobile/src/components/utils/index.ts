export * from "./array";
export { isDarkColor } from "./isDarkColor";
