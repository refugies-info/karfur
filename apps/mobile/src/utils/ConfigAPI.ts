import axios from "react-native-axios";
import Config from "../libs/getEnvironment";

export const dbURL = Config.dbUrl;
export const headers = {
  "Content-Type": "application/json",
  "site-secret": Config.siteSecret || "",
};
export const apiCaller = axios.create({
  baseURL: dbURL,
  headers,
});