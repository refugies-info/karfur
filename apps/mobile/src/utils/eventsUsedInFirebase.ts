export enum FirebaseEvent {
  REACH_END_CONTENT = "REACH_END_CONTENT",
  VALIDATE_LANGUAGE = "VALIDATE_LANGUAGE",
  VALIDATE_LOCATION = "VALIDATE_LOCATION",
  VALIDATE_AGE = "VALIDATE_AGE",
  VALIDATE_FRENCH_LEVEL = "VALIDATE_FRENCH_LEVEL",
  PROFILE_COMPLETED = "PROFILE_COMPLETED",
  PROFILE_NOT_COMPLETED = "PROFILE_NOT_COMPLETED",
  CLIC_THEME = "CLIC_THEME",
  CLIC_NEED = "CLIC_NEED",
  CLIC_CONTENT = "CLIC_CONTENT",
  CLIC_LIST = "CLIC_LIST",
  CLIC_CAROUSEL = "CLIC_CAROUSEL",
  CLIC_ACCORDION_ENGAGEMENT = "CLIC_ACCORDION_ENGAGEMENT",
  CLIC_ACCORDION = "CLIC_ACCORDION",
  CLIC_SEE_WEBSITE = "CLIC_SEE_WEBSITE",
  CLIC_SEE_MAP = "CLIC_SEE_MAP",
  SHARE_CONTENT = "SHARE_CONTENT",
  ADD_FAVORITE = "ADD_FAVORITE",
  SWITCH_LANGUAGE = "SWITCH_LANGUAGE",
  LONG_PRESS_CHANGE_LANGUAGE = "LONG_PRESS_CHANGE_LANGUAGE",
  START_VOICEOVER = "START_VOICEOVER",
  OPEN_NOTIFICATION = "OPEN_NOTIFICATION",
  CLIC_NOTIFICATION_ICON = "CLIC_NOTIFICATION_ICON",
  TOGGLE_NOTIFICATION_TYPE = "TOGGLE_NOTIFICATION_TYPE",
  TOGGLE_NOTIFICATION_THEME = "TOGGLE_NOTIFICATION_THEME",
}
