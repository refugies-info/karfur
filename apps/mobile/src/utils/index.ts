export { default as invariant } from "./invariant";
export * from "./types";
