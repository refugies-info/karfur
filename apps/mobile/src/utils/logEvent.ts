/* eslint-disable no-console */
import analytics from "@react-native-firebase/analytics";
import Config from "~/libs/getEnvironment";
import { FirebaseEvent } from "./eventsUsedInFirebase";

export const logEventInFirebase = async (eventName: FirebaseEvent, data: Record<string, any>) => {
  const { envName, debugModeFirebase } = Config;
  if (envName === "DEVELOPMENT") {
    if (debugModeFirebase) {
      console.log("Environment is dev and debug mode is activated for firebase");
    } else {
      console.log("Environment is dev and debug mode is NOT activated for firebase");
      return;
    }
  }

  if (envName && ["DEVELOPMENT", "STAGING", "PROD"].includes(envName)) {
    await analytics().logEvent(eventName, { ...data, envName });
  }
};
