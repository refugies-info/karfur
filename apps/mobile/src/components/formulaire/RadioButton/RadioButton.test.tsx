import { render } from "../../utils/tests";
import RadioButton from "./RadioButton";

describe("RadioButton snapshot test suite", () => {
  it("should render without bug", async () => {
    const test = render(<RadioButton isSelected />);
    expect(test).toMatchSnapshot();
  });
  it("should render not selected", async () => {
    const test = render(<RadioButton isSelected={false} />);
    expect(test).toMatchSnapshot();
  });
});
