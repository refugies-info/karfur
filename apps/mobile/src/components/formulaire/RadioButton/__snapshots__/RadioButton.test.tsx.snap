// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton snapshot test suite should render not selected 1`] = `
<View
  collapsable={false}
  style={
    {
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="24"
    bbWidth="24"
    fill="#161616"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={24}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4279637526,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGRect
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            height="24"
            opacity={0}
            width="24"
            x="0"
            y="0"
          />
          <RNSVGPath
            d="M12 22a10 10 0 1 1 10-10 10 10 0 0 1-10 10zm0-18a8 8 0 1 0 8 8 8 8 0 0 0-8-8z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`RadioButton snapshot test suite should render without bug 1`] = `
<View
  collapsable={false}
  style={
    {
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="24"
    bbWidth="24"
    fill="#000091"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={24}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190225,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGRect
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            height="24"
            opacity={0}
            width="24"
            x="0"
            y="0"
          />
          <RNSVGPath
            d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm0 18a8 8 0 1 1 8-8 8 8 0 0 1-8 8z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
          <RNSVGPath
            d="M12 7a5 5 0 1 0 5 5 5 5 0 0 0-5-5z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
